### PR TITLE
Default to TLS mode, unless CHAINLINK_DEV is true

### PR DESCRIPTION
**Pull request by j16r**

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -21,6 +21,9 @@ COPY --from=builder \
   /go/src/github.com/smartcontractkit/chainlink/gui/dist \
   /go/src/github.com/smartcontractkit/chainlink/gui/dist
 
+# Disable TLS... for now
+ENV CHAINLINK_DEV true
```
Comment (dimroc) [inline Dockerfile]: Do we want to default to dev mode? Could we just change `cldev` and any other consumer of our chainlink docker image to explicitly set `ENV CHAINLINK_DEV true` via command line?

This is already being done in `internal/ci/ethereum_test`, perhaps all that's left is updating `cldev`.

Comment (j16r) [reply Dockerfile]: This allows these images to work with our current automation, which does not ship any TLS key material.

Comment (dimroc) [reply Dockerfile]: Can we modify our current automation to set `CHAINLINK_DEV true`? By automation do you mean `codeship` and `kubernetes`?

Comment (dimroc) [reply Dockerfile]: I'm not strongly attached to this issue. But it seems counterintuitive to have our program default to TLS, but our docker image default to no TLS. So I'm hoping we can explicitly set `CHAINLINK_DEV true` wherever we consume the docker image via command line or environment variables.

Comment (j16r) [reply Dockerfile]: Done, see https://github.com/smartcontractkit/devnet/pull/18

**Comment by dimroc** [comment]
Waiting on https://github.com/smartcontractkit/devnet/pull/18

**Pull request merged**